core/runtime/memory.hpp

```diff
@@ -68,50 +81,25 @@ namespace kagome::runtime {
      */
     virtual std::optional<WasmSize> deallocate(WasmPointer ptr) = 0;
 
-    /**
-     * Load integers from provided address
-     */
-    virtual int8_t load8s(WasmPointer addr) const = 0;
-    virtual uint8_t load8u(WasmPointer addr) const = 0;
-    virtual int16_t load16s(WasmPointer addr) const = 0;
-    virtual uint16_t load16u(WasmPointer addr) const = 0;
-    virtual int32_t load32s(WasmPointer addr) const = 0;
-    virtual uint32_t load32u(WasmPointer addr) const = 0;
-    virtual int64_t load64s(WasmPointer addr) const = 0;
-    virtual uint64_t load64u(WasmPointer addr) const = 0;
-    virtual std::array<uint8_t, 16> load128(WasmPointer addr) const = 0;
-
-    /**
-     * Load bytes from provided address into the buffer of size n
-     * @param addr address in memory to load bytes
-     * @param n number of bytes to be loaded
-     * @return BufferView of length N
-     */
-    virtual common::BufferView loadN(WasmPointer addr, WasmSize n) const = 0;
-    /**
-     * Load string from address into buffer of size n
-     * @param addr address in memory to load bytes
-     * @param n number of bytes
-     * @return string with data
-     */
-    virtual std::string loadStr(WasmPointer addr, WasmSize n) const = 0;
-
-    /**
-     * Store integers at given address of the wasm memory
-     */
-    virtual void store8(WasmPointer addr, int8_t value) = 0;
-    virtual void store16(WasmPointer addr, int16_t value) = 0;
-    virtual void store32(WasmPointer addr, int32_t value) = 0;
-    virtual void store64(WasmPointer addr, int64_t value) = 0;
-    virtual void store128(WasmPointer addr,
-                          const std::array<uint8_t, 16> &value) = 0;
-    virtual void storeBuffer(WasmPointer addr, common::BufferView value) = 0;
-
-    /**
-     * @brief allocates buffer in memory and copies value into memory
-     * @param value buffer to store
-     * @return full wasm pointer to allocated buffer
-     */
-    virtual WasmSpan storeBuffer(common::BufferView value) = 0;
+    // endian???
+    uint32_t load32u(WasmPointer ptr) const {
+      uint32_t v;
+      memcpy(&v, view(ptr, sizeof(v)).value().data(), sizeof(v));
+      return v;
+    }
+
+    common::BufferView loadN(WasmPointer ptr, WasmSize size) const {
+      return view(ptr, size).value();
+    }
+
+    void storeBuffer(WasmPointer ptr, common::BufferView v) {
+      memcpy(view(ptr, v.size()).value().data(), v.data(), v.size());
```

Comment:
I don't like these .value()

Reply:
Wasm host api will be refactored to `outcome` later (then only wasm wrapper will throw errors).